src/input/findExistingESLintConfiguration.ts

```diff
@@ -0,0 +1,40 @@
+import { fsFileSystem } from "../adapters/fsFileSystem";
+import { GlobAsync } from "../adapters/globAsync";
+import { ResultStatus, ResultWithDataStatus } from "../types";
+import { ESLintConfiguration } from "./findESLintConfiguration";
+
+export type FindExistingESLintConfigurationDependencies = {
+    globAsync: GlobAsync;
+};
+
+export const findExistingESLintConfiguration = async (
```

Comment:
Heh, yeah, the return type of this is what I was curious about... At first, I was thinking thinking the calling code will want to know whether this is an npm package style file or an ESLint one. For example, what happens if someone does something like `--config ./packages/somewhere/package.json`, and wants this to work on the `eslintConfig` in that nested file?

But, it's really only `package.json` files that have the odd `eslintConfig` member... Maybe the writing code would always want to override an existing `eslintConfig` in the output file if that output file's name ends with `"package.json"`? What do you think